instance/models/mixins/openedx_theme.py

```diff
@@ -83,10 +83,17 @@ def get_common_settings(self, application):
         """
         Returns a dict with the simple-theme settings common to the v0 and v1 implementation.
         """
+        footer_logo_customization = {}
+        if application.footer_logo_image:
+            footer_logo_customization["FOOTER_OPENEDX_LOGO_IMAGE"] = application.footer_logo_image
+        if application.footer_logo_url:
+            footer_logo_customization["FOOTER_OPENEDX_URL"] = application.footer_logo_url
         return {
             "SIMPLETHEME_ENABLE_DEPLOY": True,
             "SIMPLETHEME_STATIC_FILES_URLS": self.get_simple_theme_static_files_settings(application),
             "EDXAPP_DEFAULT_SITE_THEME": "simple-theme",
+            "EDXAPP_LMS_ENV_EXTRA": footer_logo_customization,
+            "EDXAPP_CMS_ENV_EXTRA": footer_logo_customization
```

Comment:
I see a problem with this approach. If someone needs to override `EDXAPP_LMS_ENV_EXTRA`, footer logo customization will be lost. To avoid that, we'll have to implement some "merging" logic for this Ansible variable.

Instead, we should implement this as a new feature of https://github.com/open-craft/edx-simple-theme and generate SASS overrides. Check code of `get_simple_theme_static_files_settings` function and where it's used.

Reply:
My understanding is that all of these variables [are merged recursively already](https://github.com/open-craft/opencraft/blob/master/instance/models/openedx_appserver.py#L452).

I'm not familiar with how theming works but [the OpenEdX docs](https://edx.readthedocs.io/projects/edx-installing-configuring-and-running/en/open-release-koa.master/configuration/changing_appearance/theming/create_theme.html#understanding-which-ui-files-to-customize) state:
> To replace an image, you can override any of the images in the static/images directories for the components you are theming.

However, `FOOTER_OPENEDX_LOGO_IMAGE` defaults to `https://files.edx.org/openedx-logos/open-edx-logo-tag.png`, so I think it can't be overridden by a theme like that; theoretically, we could maybe set this to a default that can be overridden by the theme and then handle it like the other images but I'd prefer to move that into a follow-up story.

Reply:
>My understanding is that all of these variables are merged recursively already.

Oh, then, I think it's fine.

>so I think it can't be overridden by a theme like that

My idea was to override src of `img` element in [this class](https://github.com/open-craft/edx-simple-theme/blob/3d7215a31b72b8b7c3891ec5f61ea874de073fe0/lms/static/sass/theme-overrides.scss#L358-L385
) with a new SASS variable. But, after all, that would be hacky, so I'm fine with the current approach.

However, it looks like `OpenEdXThemeMixin` is not the correct place for adding these variables, since this mixing is only for   generating variables for `simple_theme`. Could you find another place to add `EDXAPP_LMS_ENV_EXTRA` and `EDXAPP_CMS_ENV_EXTRA`?

Reply:
I don't think there's a better one that exists already... Each of the configuration mixins is called in `OpenEdXInstance` to populate its respective field in `OpenEdXAppServer` and they all seem like worse options...
```python
CONFIGURATION_EXTRA_FIELDS = [
        'configuration_database_settings',
        'configuration_storage_settings',
        'configuration_theme_settings',
        'configuration_secret_keys',
        'configuration_marketing_links',
        # The extra settings should stay at the end of this list to allow manual overrides of all settings.
        'configuration_extra_settings',
    ]
```

I understand what you mean but, in this case, I don't think it's a huge issue.
I've moved these customizations to separate method and updated the docs to make it easier to understand; I hope this makes it for you.